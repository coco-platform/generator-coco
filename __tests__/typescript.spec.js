/**
 * @description - coco:typescript generator
 * @author - huang.jian <hjj491229492@hotmail.com>
 */

// packages
const path = require('path');
const fs = require('fs');
const yeomon = require('yeoman-test');
const assert = require('yeoman-assert');

describe('coco:typescript generator', () => {
  it('should output typescript configuration', () => {
    const typescript = path.resolve(__dirname, '../generators/typescript');
    const options = {
      root: './lib',
    };

    return yeomon
      .run(typescript)
      .withOptions(options)
      .then(() => {
        assert.file(['tslint.json', 'tsconfig.json']);

        const lint = fs.readFileSync('tslint.json', 'utf8');
        const compile = fs.readFileSync('tsconfig.json', 'utf8');

        expect(lint).toMatchSnapshot();
        expect(compile).toMatchSnapshot();
      });
  });
});
