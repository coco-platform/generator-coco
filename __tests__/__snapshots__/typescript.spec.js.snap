// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coco:typescript generator should output typescript configuration 1`] = `
"{
  \\"defaultSeverity\\": \\"error\\",
  \\"extends\\": [\\"tslint:recommended\\"],
  \\"jsRules\\": {},
  \\"rules\\": {
    \\"quotemark\\": [true, \\"single\\", \\"jsx-double\\"],
    \\"trailing-comma\\": [
      true,
      {
        \\"multiline\\": \\"never\\"
      }
    ]
  },
  \\"rulesDirectory\\": []
}
"
`;

exports[`coco:typescript generator should output typescript configuration 2`] = `
"{
  \\"compilerOptions\\": {
    \\"target\\": \\"esnext\\",
    \\"module\\": \\"es2015\\",
    \\"moduleResolution\\": \\"node\\",
    \\"esModuleInterop\\": true,
    \\"allowSyntheticDefaultImports\\": true,
    \\"jsx\\": \\"preserve\\",
    \\"strict\\": true,
    \\"noImplicitAny\\": true,
    \\"noUnusedLocals\\": true,
    \\"noUnusedParameters\\": true,
    \\"removeComments\\": false,
    \\"sourceMap\\": true,
    \\"skipLibCheck\\": true
  },
  \\"include\\": [\\"./lib/**/*.ts\\"],
  \\"exclude\\": [\\"node_modules\\"]
}
"
`;
